File: Exercises/Week02/src/ExcerciseScala/FireSensor.scala
package ExcerciseScala

import java.util.Random

class FireSensor(var sensorType: String) extends SensorBattery {
  var location = ""
  var triggered = false


  def this() {
    this("Fire")
  }


  def setSensorType(setSensorType : String): Unit = this.sensorType = setSensorType

  def setLocation(setLocation : String): Unit = this.location = setLocation

  def resetTriggered(): Unit = this.triggered = false




  override def isTriggered: Boolean = {
    reduceBattery(this)
    val r = new Random()
    if (r.nextInt(100) < 5){
      this.triggered = true

      return this.triggered
    }

    this.triggered
  }

  override def checkTriggered: Boolean = this.triggered

  override def getLocation = this.location

  override def getSensorType = this.sensorType

  override def getBatteryPercentage = battery

}

File: Exercises/week02/WorksheetSOLID/FireSensor.java
package Week02.WorksheetSOLID;

public class FireSensor implements Sensor {

    private boolean triggered = false;
    private String location = "";
    private String sensorType = "";
    private int batteryPercentage = 0;

    public void setTriggered(boolean triggeredAlarm) {

        triggered = triggeredAlarm;
    }

    public void setLocation(String location){

        this.location = location;
    }

    public void setSensorType(String sensorType){

        this.sensorType = sensorType;
    }

    public void setBatteryPercentage(int batteryPercentage){

        this.batteryPercentage = batteryPercentage;
    }

    @Override
    public boolean isTriggered() {
        return triggered;
    }

    @Override
    public String getLocation() {
        return location;
    }

    @Override
    public String getSensorType() {
        return sensorType;
    }

    @Override
    public double getBatteryPercentage() {
        return batteryPercentage;
    }
}
